frontend/pages/dashboard/[id].js

```diff
@@ -370,46 +373,61 @@ export default function Dashboard() {
     );
 
     // Checking if any of the secret keys start with a number - if so, don't do anything
-    const nameErrors = !Object.keys(obj).map(key => !isNaN(key.charAt(0))).every(v => v === false);
-    const duplicatesExist = data?.map(item => item[2]).filter((item, index) => index !== data?.map(item => item[2]).indexOf(item)).length > 0;
+    const nameErrors = !Object.keys(obj)
+      .map((key) => !isNaN(key.charAt(0)))
+      .every((v) => v === false);
+    const duplicatesExist =
+      data
+        ?.map((item) => item[2])
+        .filter(
+          (item, index) => index !== data?.map((item) => item[2]).indexOf(item)
+        ).length > 0;
 
     if (nameErrors) {
-      console.log("Solve all name errors first!");
-    } else if (duplicatesExist) {
-      console.log("Remove the duplicated entries first!");
-    } else {
-      // Once "Save changes is clicked", disable that button
-      setButtonReady(false);
-      pushKeys({obj, workspaceId: router.query.id, env});
-
-      /**
-       * Check which integrations are active for this project and environment
-       * If there are any, update environment variables for those integrations
-       */
-      let integrations = await getWorkspaceIntegrations({
-        workspaceId: router.query.id,
+      return createNotification({
+        text: "Solve all name errors first!",
+        type: "error",
       });
-      integrations.map(async (integration) => {
-        if (
-          envMapping[env] == integration.environment &&
-          integration.isActive == true
-        ) {
-          let objIntegration = Object.assign(
-            {},
-            ...data.map((row) => ({ [row[2]]: row[3] }))
-          );
-          await pushKeysIntegration({
-            obj: objIntegration,
-            integrationId: integration._id,
-          });
-        }
+    }
+
+    if (duplicatesExist) {
+      return createNotification({
+        text: "Your secrets weren't saved; please fix the conflicts first.",
+        type: "error",
       });
+    }
 
-      // If this user has never saved environment variables before, show them a prompt to read docs
-      if (!hasUserEverPushed) {
-        setCheckDocsPopUpVisible(true);
-        await registerUserAction({ action: "first_time_secrets_pushed" });
+    // Once "Save changed is clicked", disable that button
+    setButtonReady(false);
+    pushKeys(obj, router.query.id, env);
```

Comment:
@asharonbaltazar  I think this line should be `pushKeys({obj, workspaceId: router.query.id, env});` I tested and otherwise it seems to break. Would you pleas be able to change it?

Reply:
Yes, no prob; this isn't related to my changes

Reply:
The joys of JavaScript!